includes/classes/Elasticsearch.php

```diff
@@ -1321,28 +1457,15 @@ public function create_pipeline( $id, $args ) {
 	}
 
 	/**
-	 * Query logging.
-	 *
-	 * If EP_QUERY_LOG is defined, use its value to control if
-	 * query logging is enabled. If not, only enable it if WP_DEBUG
-	 * or WP_EP_DEBUG are enabled.
-	 *
-	 * Calls action 'ep_add_query_log' if you want to access the
-	 * query outside of the ElasticPress plugin. This runs regardless
-	 * of debug settings.
+	 * Query logging. Don't log anything to the queries property when
+	 * WP_DEBUG is not enabled. Calls action 'ep_add_query_log' if you
+	 * want to access the query outside of the ElasticPress plugin. This
+	 * runs regardless of debufg settings.
 	 *
 	 * @param array $query Query to log.
 	 */
 	protected function add_query_log( $query ) {
-		$log_enabled = false;
-
-		if ( defined( 'EP_QUERY_LOG' ) ) {
```

Comment:
Hmm, this is a change from upstream EP @netsuso - what do you suggest? Should we use `WP_EP_DEBUG` instead?

Reply:
If I'm understanding the upstream change correctly, `EP_QUERY_LOG` is entirely removed and logging will only be enabled if `WP_DEBUG` or `WP_EP_DEBUG` is set, so it should work the same and we can actually remove that constant from class-search.php

Reply:
That `EP_QUERY_LOG` has never existed upstream. I added it in our fork: https://github.com/Automattic/ElasticPress/commit/ef842a0d

So that diff above is precisely reverting my commit. I wouldn't like to rely on `WP_DEBUG` or `WP_EP_DEBUG`, as they are set elsewhere for different reasons, and I don't want to disable them just to avoid logging.

I don't have a clue about why this revert has been detected, what have you done to import recent changes from upstream?

Reply:
That makes more sense 😄 . @pschoffer brought synced down some of Elasticsearch.php in https://github.com/Automattic/ElasticPress/pull/81/commits/5c1bce11bd562cd8bd6e8a1426f311fe34734a42 - maybe this change was an oversight?

Reply:
Reverted those changes in 42199573, great catch @netsuso!

Reply:
Awesome, thank you! This way we are avoiding some unexpected OOM in search commands, I wouldn't have enjoyed seeing them again :sweat_smile: